test/net/sourceforge/kolmafia/textui/ParserTest.java

```diff
@@ -1333,6 +1373,20 @@ public void testScriptValidity(ScriptData script) {
 
     // This will fail if an exception is thrown.
     script.parser.parse();
-    assertEquals(((ValidScriptData) script).tokens, script.parser.getTokensContent(), script.desc);
+    assertEquals(((ValidScriptData) script).tokens, getTokensContent(script.parser), script.desc);
+    if (((ValidScriptData) script).positions != null) {
+      assertEquals(
+          ((ValidScriptData) script).positions, getTokensPosition(script.parser), script.desc);
+    }
+  }
+
+  public static List<String> getTokensContent(final Parser parser) {
+    return parser.getTokens().stream().map(token -> token.content).collect(Collectors.toList());
+  }
+
+  public static List<Integer> getTokensPosition(final Parser parser) {
+    return parser.getTokens().stream()
+        .map(token -> (token.getStart().getCharacter() + 1) * 10 + token.getStart().getLine() + 1)
```

Comment:
This broadly feels wrong.

Character 8, line 5 -> 96? Even ignoring the change from 0-indexing to 1-indexing, a position in a file is often serialized as line,character, not character,line.

This also behaves poorly for text that's >= 10 lines long. (I recognize that's probably why you serialized it this way, since none of these cases are 10+ lines long.)

I'd prefer an explicit pair of integers, or failing that, a string.